### PR TITLE
GlusterFS: provide default for groups.oo_glusterfs_to_config in with_items

### PR DESCRIPTION
**Pull request by jarrpa**
This will hopefully resolve the problems encountered [here](https://github.com/openshift/openshift-ansible/pull/3793#issuecomment-294859858).

### CODE REVIEW

**Comment by sdodson** [comment]
bot, add author to whitelist

**Comment by sdodson** [comment]
aos-ci-test

**Comment by sdodson** [comment]
[test]

**Comment by sdodson** [comment]
LGTM, but saw you and @kwoodson discussing so assigned him to review

**Comment by jarrpa** [comment]
I'd also like to wait for @smarterclayton to weigh in and let us know if it resolves the issue he was seeing.

File: roles/openshift_storage_glusterfs/tasks/glusterfs_deploy.yml [hunk 1]
```diff
@@ -102,6 +102,6 @@
   until:
   - "glusterfs_pods.results.results[0]['items'] | count > 0"
   # There must be as many pods with 'Ready' staus  True as there are nodes expecting those pods
-  - "glusterfs_pods.results.results[0]['items'] | oo_collect(attribute='status.conditions') | oo_collect(attribute='status', filters={'type': 'Ready'}) | map('bool') | select | list | count == groups.oo_glusterfs_to_config | count"
+  - "glusterfs_pods.results.results[0]['items'] | oo_collect(attribute='status.conditions') | oo_collect(attribute='status', filters={'type': 'Ready'}) | map('bool') | select | list | count == glusterfs_nodes | count"
```
Comment (kwoodson) [inline roles/openshift_storage_glusterfs/tasks/glusterfs_deploy.yml]: Might help to put parenthesis here to show what is desired.

Comment (jarrpa) [inline roles/openshift_storage_glusterfs/tasks/glusterfs_deploy.yml]: I'll consider that for another PR. :)

**Comment by smarterclayton** [comment]
Still failing - why do you assume glusterfs_nodes is set?

```
skipping: [ci-claytontest-ig-m-tb2l] => (item=ci-claytontest-ig-n-kdr1)

TASK [openshift_storage_glusterfs : Delete pre-existing GlusterFS config] ******
Tuesday 18 April 2017  16:52:29 +0000 (0:00:00.210)       0:39:53.874 *********
fatal: [ci-claytontest-ig-m-tb2l]: FAILED! => {"failed": true, "msg": "'glusterfs_nodes' is undefined"}
```

**Comment by jarrpa** [comment]
@smarterclayton Because I explicitly set it here: https://github.com/jarrpa/openshift-ansible/blob/77091384ffb7a61fc467a6fc09dd367a37a5f1cc/roles/openshift_storage_glusterfs/tasks/main.yml#L3

**Comment by jarrpa** [comment]
@smarterclayton The question I have is why is your test hitting this issue while neither the ones here nor mine are? Best we can tell EVERYTHING in this roles should be explicitly skipped if no ansible group `glusterfs` exists.

**Comment by jarrpa** [comment]
After a short troubleshooting session with @smarterclayton, found out that this is a symptom of https://github.com/ansible/ansible/issues/13598 . Urgh. New patch pushed.

**Comment by smarterclayton** [comment]
Failing due to:

```
            "key": "oo_nodes_deployment_type_origin"
        },
        "module_name": "group_by"
    }
}
ok: [ci-claytontest-ig-m-xch7] => {
    "add_group": "oo_nodes_deployment_type_origin",
    "changed": false,
    "invocation": {
        "module_args": {
            "key": "oo_nodes_deployment_type_origin"
        },
        "module_name": "group_by"
    }
}
ok: [ci-claytontest-ig-n-v54g] => {
    "add_group": "oo_nodes_deployment_type_origin",
    "changed": false,
    "invocation": {
        "module_args": {
            "key": "oo_nodes_deployment_type_origin"
        },
        "module_name": "group_by"
    }
}
ok: [ci-claytontest-ig-n-zwd1] => {
    "add_group": "oo_nodes_deployment_type_origin",
    "changed": false,
    "invocation": {
        "module_args": {
            "key": "oo_nodes_deployment_type_origin"
        },
        "module_name": "group_by"
    }
}
ERROR! the field 'hosts' is required but was not set
```

I don't have any instances tagged with glusterfs.  It *looks* like that's a failure condition.

**Comment by jarrpa** [comment]
@smarterclayton ....I'm afraid I don't understand how that's possible. I thought tags were just informational and applied to tasks, not hosts? What am I missing? And WHY is this still only on origin-gce??

**Comment by smarterclayton** [comment]
Ah, it could have been in the first play inside of openshift-glusterfs/config.yml:

```
- name: Open firewall ports for GlusterFS
  hosts: oo_glusterfs_to_config
  vars:
    os_firewall_allow:
    - service: glusterfs_sshd
      port: "2222/tcp"
    - service: glusterfs_daemon
      port: "24007/tcp"
    - service: glusterfs_management
      port: "24008/tcp"
    - service: glusterfs_bricks
      port: "49152-49251/tcp"
  roles:
  - os_firewall
```

If `oo_glusterfs_to_config` is empty, is the play skipped?

**Comment by smarterclayton** [comment]
Or should this also have:

```
    when: groups.oo_glusterfs_to_config | default([]) | count > 0
```

**Comment by smarterclayton** [comment]
@kwoodson can you think of reasons why I would get a hard failure there?

**Comment by abutcher** [comment]
aos-ci-test

**Comment by smarterclayton** [comment]
This passed on my run the second time - not sure what was causing the issue the first time.  This is good to go for me.

**Comment by abutcher** [comment]
flake https://github.com/openshift/origin/issues/13705

**Comment by abutcher** [comment]
[test]

**Comment by sdodson** [comment]
[merge]

**Comment by abutcher** [comment]
flake https://github.com/openshift/origin/issues/13814

**Comment by abutcher** [comment]
[merge]

**Comment by abutcher** [comment]
flake https://github.com/openshift/origin/issues/13846

**Comment by abutcher** [comment]
[merge]

**Pull request merged**